crates/solidity/testing/sanctuary/src/tests.rs

```diff
@@ -100,21 +104,45 @@ pub fn run_test(file: &SourceFile, events: &Events) -> Result<()> {
 
     let parser = Parser::create(version.clone())?;
     let output = parser.parse(NonterminalKind::SourceUnit, &source);
+    let source_id = file.path.strip_repo_root()?.unwrap_str();
 
     if output.is_valid() {
-        events.test(TestOutcome::Passed);
-        return Ok(());
-    }
+        let mut outcome = TestOutcome::Passed;
```

Comment:
nit: I suggest moving the new binding code to a separate `check_bindings()` or maybe combine it with `create_bindings()` if you like, to keep `run_test()` focused on reporting results..

Reply:
Makes sense. I made a couple of attempts to refactor this function to extract different parts, but I didn't like the results, especially because you have to send almost all locals as arguments. I'll make another pass.